### ISSUE TITLE
Can't run Consul agent on Docker Swarm nodes

### ISSUE THREAD

**Issue by dayglojesus** [moby/moby#25945]
Hi,

I've run across an issue trying to run Consul on Swarm nodes.

Swarm operations like `join`, `promote`, `demote` all work, but Leader election doesn't.

While the Consul agent is running, every 10 seconds a log message like this appears:

`Aug 23 05:00:31 i-d34db33f dockerd[17561]: time="2016-08-23T05:00:31.155111218Z" level=warning msg="2016/08/23 05:00:31 [ERR] memberlist: failed to receive: EOF from=10.170.17.174:60120\n"`

If I induce failure on the Leader node by stopping the Docker daemon, we get a failed election and neither of the remaining Manager nodes take leadership.

Stopping the Consul agent on the Manager nodes resolves this issue and Leader election resumes normally and the ERR messages vanish.

**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.0
Storage Driver: devicemapper
 Pool Name: docker-253:3-20-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 632.7 MB
 Data Space Total: 107.4 GB
 Data Space Available: 3.375 GB
 Metadata Space Used: 966.7 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /opt/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /opt/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay null bridge host
Swarm: active
 NodeID: 25qxub4lnt92xi0kn7kvcvrkl
 Is Manager: true
 ClusterID: 3omuj7snd6ms79lak6grqo25x
 Managers: 3
 Nodes: 6
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot interval: 10000
  Heartbeat tick: 1
  Election tick: 3
 Dispatcher:
  Heartbeat period: 5 seconds
 CA configuration:
  Expiry duration: 3 months
 Node Address: 10.170.17.171
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.22.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.796 GiB
Name: i-d34db33f
ID: R44H:4MH6:5S35:ZCOH:AJI7:BYQG:UTSL:MP2T:PKT4:RIE6:JBIT:Q66K
Docker Root Dir: /opt/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
- 6 node swarm (3 managers) running in AWS, single AZ

**Steps to reproduce the issue:**
1. Init a small swarm: 6 nodes, 3 managers
2. Install Consul on all nodes and configure appropriately
3. Induce failure to the Swarm Leader

**Describe the results you received:**
- Swarm fails to elect a new Leader
- log messages indicate communications issues

**Describe the results you expected:**
- Consul and Docker in Swarm mode should operate properly running in parallel

**Additional information you deem important (e.g. issue happens only occasionally):**
- Issue seems simple to reproduce; I am guessing the Docker's internal use of Consul is the source of the issue.

**Comment by dayglojesus** [comment]
**Update:** this issue does seem to be intermittent or some form entropic behaviour. After reinitializing the Swarm, Leader election appears to be operating normally, despite the ERR messages being logged...

`Aug 23 06:10:08 i-de4db33f dockerd[28644]: time="2016-08-23T06:10:08.745456688Z" level=warning msg="2016/08/23 06:10:08 [ERR] memberlist: failed to receive: EOF from=10.170.17.171:41134\n"`

The Swarm had been operating for a few weeks before I noticed the Leader election process failing.

Now, I'm scratching my head...

**Comment by justincormack** [comment]
Can you see if upgrading to 1.12.1 fixes it? There were some swarm mode
networking fixes.

**Comment by dayglojesus** [comment]
@justincormack Thanks for your reply. I've installed version 1.12.1 across all nodes. I can no longer reproduce the reported issue anymore, however the new Swarm has only been running for about 24 hours. 

Note: the log messages continue to roll in, the same message **once per second** on the Leader node, once every 10 seconds on the other available Manager nodes.

I will continue to monitor the behaviour. Thanks again and please let me know if I can provide any further information.

**Comment by justincormack** [comment]
Is this still an issue?

**Comment by dayglojesus** [comment]
Sorry of the lack of follow-up, yes and no. The log messages persist in the manner described above, but I cannot reproduce the Leader election failure. The Swarm has been functional since my last report. Thanks for the follow-up reminder.

**Comment by thaJeztah** [comment]
Let me close this ticket for now, as it looks like it went stale.
